### DISCUSSION TITLE
Pass logs from `xcodebuild` to gym's exception when build fails

### DISCUSSION THREAD

**Discussion by sanju-naik** [fastlane/fastlane#16878] [Ideas]
Can we pass logs from `xcodebuild` to gym's error/exception object when build fails. This would help us to parse the actual error from build logs and surface it to the end of Fastlane logs on our CI & also send it via Slack. 

Actually logs from `xcodebuild` are collected in `output` variable and are being sent to `build_failure!` method here - https://github.com/fastlane/fastlane/blob/a68011c570bd429e91d5adeb1c363a0dce49ec82/gym/lib/gym/error_handler.rb#L73 Now we just need to add a property on `FastlaneBuildFailure` say `error_info` similar to `FastlaneError` - https://github.com/fastlane/fastlane/blob/a68011c570bd429e91d5adeb1c363a0dce49ec82/fastlane_core/lib/fastlane_core/ui/errors/fastlane_error.rb#L7 and pass output logs to it while raising exception here: https://github.com/fastlane/fastlane/blob/a68011c570bd429e91d5adeb1c363a0dce49ec82/fastlane_core/lib/fastlane_core/ui/interface.rb#L162 . 

Let me know your thoughts on this, I am happy to help with a pull request on this.

**Comment by eliw00d** [comment]
This would be nice and I was actually surprised when I ran into this error:
```
An error occurred while executing the `error` block:
undefined method `error_info' for #<FastlaneCore::Interface::FastlaneBuildFailure:0x00007fd63843d698>
```
Since the [docs](https://docs.fastlane.tools/advanced/Fastfile/) tell you to use `error_info`.
